Kaetram/Kaetram-Open
# Move the application config to env file

### Motivation

Closes #44

Environment configuration has been moved to `.env` files to make the deployment more flexible.

I tried to deploy the client to netlify - successfully, and the server to heroku - no luck unfortunately, I need to spend some more time on it, I was facing some out of memory errors:

```
2020-07-02T08:36:14.270770+00:00 app[web.1]: ==== JS stack trace =========================================
2020-07-02T08:36:14.270770+00:00 app[web.1]: 
2020-07-02T08:36:14.270771+00:00 app[web.1]: 0: ExitFrame [pc: 0x134e879]
2020-07-02T08:36:14.270771+00:00 app[web.1]: 1: StubFrame [pc: 0x134f753]
2020-07-02T08:36:14.270771+00:00 app[web.1]: Security context: 0x1a04c5c00919 <JSObject>
2020-07-02T08:36:14.270772+00:00 app[web.1]: 2: writeText(aka writeText) [0x6257c963a29] [/app/packages/server/node_modules/typescript/lib/tsc.js:~12774] [pc=0x3cd1e8d4e372](this=0x39596cb004d1 <undefined>,0x019611972ad9 <String[4]: 3777>)
2020-07-02T08:36:14.270774+00:00 app[web.1]: 3: write [0x6257c963a69] [/app/packages/server/node_modules/typescript/lib/tsc.js:~12784] [pc=0x3cd1e8d4d702](this=0x06257c963aa9...
2020-07-02T08:36:14.270774+00:00 app[web.1]: 
2020-07-02T08:36:14.270781+00:00 app[web.1]: FATAL ERROR: Ineffective mark-compacts near heap limit Allocation failed - JavaScript heap out of memory
2020-07-02T08:36:14.270983+00:00 app[web.1]: 
2020-07-02T08:36:14.281016+00:00 app[web.1]: Writing Node.js report to file: report.20200702.083614.21.0.001.json
2020-07-02T08:36:14.281017+00:00 app[web.1]: Node.js report completed
2020-07-02T08:36:14.281508+00:00 app[web.1]: 1: 0x9bcb80 node::Abort() [node]
2020-07-02T08:36:14.282081+00:00 app[web.1]: 2: 0x9bdd16 node::OnFatalError(char const*, char const*) [node]
2020-07-02T08:36:14.282573+00:00 app[web.1]: 3: 0xb1948e v8::Utils::ReportOOMFailure(v8::internal::Isolate*, char const*, bool) [node]
2020-07-02T08:36:14.283142+00:00 app[web.1]: 4: 0xb19809 v8::internal::V8::FatalProcessOutOfMemory(v8::internal::Isolate*, char const*, bool) [node]
2020-07-02T08:36:14.283706+00:00 app[web.1]: 5: 0xcca1f5  [node]
2020-07-02T08:36:14.284232+00:00 app[web.1]: 6: 0xcca866 v8::internal::Heap::RecomputeLimits(v8::internal::GarbageCollector) [node]
2020-07-02T08:36:14.284765+00:00 app[web.1]: 7: 0xcd609a v8::internal::Heap::PerformGarbageCollection(v8::internal::GarbageCollector, v8::GCCallbackFlags) [node]
2020-07-02T08:36:14.285334+00:00 app[web.1]: 8: 0xcd6fa5 v8::internal::Heap::CollectGarbage(v8::internal::AllocationSpace, v8::internal::GarbageCollectionReason, v8::GCCallbackFlags) [node]
2020-07-02T08:36:14.285879+00:00 app[web.1]: 9: 0xcd9a68 v8::internal::Heap::AllocateRawWithRetryOrFail(int, v8::internal::AllocationType, v8::internal::AllocationAlignment) [node]
2020-07-02T08:36:14.286427+00:00 app[web.1]: 10: 0xca1597 v8::internal::Factory::NewFillerObject(int, bool, v8::internal::AllocationType) [node]
2020-07-02T08:36:14.287072+00:00 app[web.1]: 11: 0xfc9708 v8::internal::Runtime_AllocateInYoungGeneration(int, unsigned long*, v8::internal::Isolate*) [node]
2020-07-02T08:36:14.287792+00:00 app[web.1]: 12: 0x134e879  [node]
2020-07-02T08:36:14.316868+00:00 app[web.1]: Aborted
```

Anyway, I tested the changes locally, both dev and prod modes and they work fine.

Additionally I replaced `bcrypt` with `bcryptjs` as it was causing some build errors on heroku.

Let me know if sth is unclear as there is quite a lot of modifications.

**How to test (feature)**

Clone fresh version of the repository, remove `-dist` postfixes from `.env` files, provide the proper configuration
- run the client and the server with the development commands
- build the client and the server then run these builds.

Result: in both cases, the application should work as before.
